src/lib/autocomplete/autocomplete-trigger.ts

```diff
@@ -138,14 +149,33 @@ export class MatAutocompleteTrigger implements ControlValueAccessor, OnDestroy {
   /* The autocomplete panel to be attached to this trigger. */
   @Input('matAutocomplete') autocomplete: MatAutocomplete;
 
+  /** Stream of clicks outside of the autocomplete panel. */
+  private _outsideClickStream: Observable<any>;
+
   constructor(private _element: ElementRef, private _overlay: Overlay,
               private _viewContainerRef: ViewContainerRef,
               private _zone: NgZone,
               private _changeDetectorRef: ChangeDetectorRef,
               @Inject(MAT_AUTOCOMPLETE_SCROLL_STRATEGY) private _scrollStrategy,
               @Optional() private _dir: Directionality,
               @Optional() @Host() private _formField: MatFormField,
-              @Optional() @Inject(DOCUMENT) private _document: any) {}
+              @Optional() @Inject(DOCUMENT) document: any) {
+    this._nextStable = first.call(_zone.onStable);
```

Comment:
Since you're wrapping it in a `first`, it means that this will only ever fire once. There a couple of places below that assume that it'll fire multiple times.

Reply:
Not true. It's one of more confusing aspects of observables. They are more like functions than promises. On each new subscription, it will wait for an onStable emission and complete.

See https://github.com/angular/material2/pull/7776#discussion_r144660550